apps/wing/src/commands/compile.ts

```diff
@@ -249,6 +251,14 @@ export async function compile(entrypoint: string, options: ICompileOptions): Pro
     return process.exit(1);
   }
 
+  // clean up before
+  await Promise.all([
+    rm(synthDir, { recursive: true, force: true }),
+    mkdir(synthDir, { recursive: true }),
+  ]);
```

Comment:
Is there a race condition here? maybe it's impossible, but what if mkdir runs before rm?

```suggestion
  // clean up before
  await rm(synthDir, { recursive: true, force: true });
  await mkdir(synthDir, { recursive: true });
```